### PR TITLE
DOCSP-38279: graalvm metadata

### PR DESCRIPTION
**Pull request by rustagir**
https://jira.mongodb.org/browse/DOCSP-38279

Adds an entry for JVM v5.2 release. will be included in the whats new pages for all 5 drivers.

### CODE REVIEW

**Review by mongoKart — APPROVED** [review]
LGTM w/ 1 suggestion

File: dbx/jvm/v5.2-wn-items.rst [hunk 1]
```diff
@@ -16,4 +16,20 @@
...
+  <https://www.graalvm.org/latest/reference-manual/native-image/metadata/>`__
+  in the GraalVM documentation.
+
+  This change does not add ``libjnidispatch`` and ``libmongocrypt``
+  resource entries, because adding specify entries for all supported
+  platforms (targets) significantly affects the size of
+  native executables built using GraalVM Native Image. View this sample
+  `resource-config.json
+  <https://github.com/mongodb/mongo-java-driver/blob/master/graalvm-native-image-app/src/main/resources/META-INF/native-image/resource-config.json>`__
+  file in the JVM driver GitHub to see how to specify these entries explicitly
```
Comment (stIncMale) [inline dbx/jvm/v5.2-wn-items.rst:34]:
```suggestion
  file in the driver GitHub repository to see how to specify these entries explicitly
```

**Review by stIncMale — APPROVED** [review]

File: dbx/jvm/v5.2-wn-items.rst [hunk 2]
```diff
@@ -16,4 +16,20 @@
...
 - When a sharded cluster operation is unsuccessful, the driver avoids selecting
   the same ``mongos`` server for operation retry attempts if other ``mongos``
-  servers are available.  
\ No newline at end of file
+  servers are available.
+
+- Adds reachability metadata needed when your application uses GraalVM
+  Native Image. You can use this metadata instead of
+  collecting reachability metadata needed when using the driver
+  libraries yourself. To learn more, see `Reachability Metadata
```
[the remark this answers is not in the document]
Comment (stIncMale) [reply dbx/jvm/v5.2-wn-items.rst]: ✅

File: dbx/jvm/v5.2-wn-items.rst [hunk 3]
```diff
@@ -16,4 +16,20 @@
...
+  <https://www.graalvm.org/latest/reference-manual/native-image/metadata/>`__
+  in the GraalVM documentation.
+
+  This change does not add ``libjnidispatch`` and ``libmongocrypt``
+  resource entries, because adding specify entries for all supported
```
[the remark this answers is not in the document]
Comment (stIncMale) [reply dbx/jvm/v5.2-wn-items.rst]: ✅

File: dbx/jvm/v5.2-wn-items.rst (hunk 1 above)
Comment (stIncMale) [reply dbx/jvm/v5.2-wn-items.rst]: ✅

**Pull request merged**